src/compiler/emitter.ts

```diff
@@ -4395,7 +4403,23 @@ module ts {
                     emitComputedPropertyName(<ComputedPropertyName>memberName);
                 }
                 else {
-                    write(".");
+                    // For ES6 and above, we want to emit memberName by itself without prefix ".",

```

Comment:
Actually, my expectation is that emitMemberAccessForPropertyName would not get called for above ES6. The reason is that in ES6, you are not emitting a member access. So I feel like there should be other logic for ES6 that does not go through this code path.